core/src/main/scala/org/apache/spark/scheduler/BlacklistStrategy.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.collection.mutable
+
+import org.apache.spark.Logging
+import org.apache.spark.SparkConf
+import org.apache.spark.util.SystemClock
+
+/**
+ * The interface to determine executor blacklist and node blacklist.
+ */
+trait BlacklistStrategy {
+  val expireTimeInMillisecond: Long
+
+  // TaskId is optional here because the BlacklistStrategy could be unrelated to TaskId
```

Comment:
Not exactly. If TaskID unrelated strategy is applied, the caller should be valid to not provide taskID, in this case, I think input "None" is better than "null"? The logic is:
1. Caller provide TaskID + ID related Strategy => works fine
2. Caller provide TaskID + ID unrelated Strategy => print warn log to announce that provided ID will be ignored, and works as Strategy defined.
3. Caller NOT provide TaskID + ID related Strategy => Strategy can decide the behavior, eg: return blacklisted executor of all IDs or return empty, or even throw exception
4. Caller NOT provide TaskID + ID unrelated Strategy => works fine